kratos/factories/standard_builder_and_solver_factory.cpp

```diff
@@ -0,0 +1,62 @@
+        typedef LinearSolver<SpaceType, LocalSpaceType> LinearSolverType;
+
+        typedef BuilderAndSolver<SpaceType,  LocalSpaceType, LinearSolverType> BuilderAndSolverType;
+        typedef ResidualBasedEliminationBuilderAndSolver<SpaceType,  LocalSpaceType, LinearSolverType> ResidualBasedEliminationBuilderAndSolverType;
+        typedef ResidualBasedEliminationBuilderAndSolverWithConstraints<SpaceType,  LocalSpaceType, LinearSolverType> ResidualBasedEliminationBuilderAndSolverWithConstraintsType;
+        typedef ResidualBasedBlockBuilderAndSolver<SpaceType,  LocalSpaceType, LinearSolverType> ResidualBasedBlockBuilderAndSolverType;
+        typedef ResidualBasedBlockBuilderAndSolverWithConstraints<SpaceType,  LocalSpaceType, LinearSolverType> ResidualBasedBlockBuilderAndSolverWithConstraintsType;
+
+        //NOTE: here we must create persisting objects for the builder and solvers
+//         static auto BuilderAndSolverFactory = StandardBuilderAndSolverFactory<BuilderAndSolverType, LinearSolverType, BuilderAndSolverType>();
+        static auto ResidualBasedEliminationBuilderAndSolverFactory = StandardBuilderAndSolverFactory<BuilderAndSolverType, LinearSolverType, ResidualBasedEliminationBuilderAndSolverType>();
+        static auto ResidualBasedEliminationBuilderAndSolverWithConstraintsFactory = StandardBuilderAndSolverFactory<BuilderAndSolverType, LinearSolverType, ResidualBasedEliminationBuilderAndSolverWithConstraintsType>();
+        static auto ResidualBasedBlockBuilderAndSolverFactory = StandardBuilderAndSolverFactory<BuilderAndSolverType, LinearSolverType, ResidualBasedBlockBuilderAndSolverType>();
+        static auto ResidualBasedBlockBuilderAndSolverWithConstraintsFactory = StandardBuilderAndSolverFactory<BuilderAndSolverType, LinearSolverType, ResidualBasedBlockBuilderAndSolverWithConstraintsType>();
+
+        // Registration of convergence solvers
+//         KRATOS_REGISTER_BUILDER_AND_SOLVER("BuilderAndSolver", BuilderAndSolverFactory);
+        KRATOS_REGISTER_BUILDER_AND_SOLVER("ResidualBasedEliminationBuilderAndSolver", ResidualBasedEliminationBuilderAndSolverFactory);
+        KRATOS_REGISTER_BUILDER_AND_SOLVER("elimination_builder_and_solver", ResidualBasedEliminationBuilderAndSolverFactory);
+        KRATOS_REGISTER_BUILDER_AND_SOLVER("ResidualBasedEliminationBuilderAndSolverWithConstraints", ResidualBasedEliminationBuilderAndSolverWithConstraintsFactory);
+        KRATOS_REGISTER_BUILDER_AND_SOLVER("elimination_builder_and_solver_with_constraints", ResidualBasedEliminationBuilderAndSolverWithConstraintsFactory);
+        KRATOS_REGISTER_BUILDER_AND_SOLVER("ResidualBasedBlockBuilderAndSolver", ResidualBasedBlockBuilderAndSolverFactory);
+        KRATOS_REGISTER_BUILDER_AND_SOLVER("block_builder_and_solver", ResidualBasedBlockBuilderAndSolverFactory);
+        KRATOS_REGISTER_BUILDER_AND_SOLVER("ResidualBasedBlockBuilderAndSolverWithConstraints", ResidualBasedBlockBuilderAndSolverWithConstraintsFactory);
```

Comment:
1. this is outdated
2. why do you register with two different names? => I would say only use the style-guide consistent name

Reply:
1- Life is always changing, never stops. Chaos is what awaits us
2- Because (retrocompatibility¿?)

Reply:
2. it is a new feature, why do you need backwardscompatibility?